src/lsm/tree_fuzz.zig

```diff
@@ -0,0 +1,393 @@
+    Key,
+    Key.Value,
+    Key.compare_keys,
+    Key.key_from_value,
+    Key.sentinel_key,
+    Key.tombstone,
+    Key.tombstone_from_key,
+);
+const Tree = @import("tree.zig").TreeType(Table, Storage, @typeName(Table) ++ "_test");
+
+const Grid = GridType(Storage);
+const SuperBlock = vsr.SuperBlockType(Storage);
+
+const Key = packed struct {
+    id: u64 align(@alignOf(u64)),
+
+    const Value = packed struct {
+        id: u64,
+        value: u63,
+        tombstone: u1 = 0,
+    };
+
+    inline fn compare_keys(a: Key, b: Key) std.math.Order {
+        return std.math.order(a.id, b.id);
+    }
+
+    inline fn key_from_value(value: *const Key.Value) Key {
+        return Key{ .id = value.id };
+    }
+
+    const sentinel_key = Key{
+        .id = std.math.maxInt(u64),
+    };
+
+    inline fn tombstone(value: *const Key.Value) bool {
+        return value.tombstone != 0;
+    }
+
+    inline fn tombstone_from_key(key: Key) Key.Value {
+        return Key.Value{
+            .id = key.id,
+            .value = 0,
+            .tombstone = 1,
+        };
+    }
+};
+
+const FuzzOp = union(enum) {
+    // TODO Test checkpointing, range queries.
+    compact,
+    put: Key.Value,
+    remove: Key.Value,
+    get: Key,
+};
+
+const Environment = struct {
+    const cluster = 32;
+    const replica = 4;
+    // TODO Is this appropriate for the number of fuzz_ops we want to run?
+    const size_max = vsr.Zone.superblock.size().? + vsr.Zone.wal.size().? + 1024 * 1024 * 1024;
+
+    const node_count = 1024;
+    // This is the smallest size that set_associative_cache will allow us.
+    const cache_entries_max = 2048;
+    const batch_size_max = config.message_size_max - @sizeOf(vsr.Header);
+    const commit_entries_max = @divFloor(batch_size_max, @sizeOf(Key.Value));
+    const tree_options = Tree.Options{
+        .commit_entries_max = commit_entries_max,
+    };
+
+    const State = enum {
+        uninit,
+        init,
+        formatted,
+        superblock_open,
+        tree_open,
+        tree_compacting,
+        tree_lookup,
+    };
+
+    state: State,
+    storage: *Storage,
+    message_pool: MessagePool,
+    superblock: SuperBlock,
+    superblock_context: SuperBlock.Context,
+    grid: Grid,
+    node_pool: NodePool,
+    cache: Tree.TableMutable.ValuesCache,
+    tree: Tree,
+    // We need @fieldParentPtr() of tree, so we can't use an optional Tree.
+    tree_exists: bool,
+    lookup_context: Tree.LookupContext = undefined,
+    lookup_value: ?*const Key.Value = null,
+
+    fn init(env: *Environment, storage: *Storage) !void {
+        env.state = .uninit;
+
+        env.storage = storage;
+        errdefer env.storage.deinit(allocator);
+
+        env.message_pool = try MessagePool.init(allocator, .replica);
+        errdefer env.message_pool.deinit(allocator);
+
+        env.superblock = try SuperBlock.init(allocator, env.storage, &env.message_pool);
+        env.superblock_context = undefined;
```

Comment:
Nit: I think you could just change the field declaration to `superblock_context: SuperBlock.Context = undefined,` like we do for `lookup_context` instead. This also keeps the `SuperBlock` `init()`/`deinit()` adjacent in the code.